### PR TITLE
Feat/phone email link enhancements

### LINKED ISSUE #1055: Update phone/email links: underline, format international numbers
Need to improve the design and interaction for phone and email links by separating them from other links and modifying the behavior when clicking on the phone edit icon. Ticket linked to [#937](https://github.com/twentyhq/twenty/issues/937), [#938](https://github.com/twentyhq/twenty/issues/938) and [#883](https://github.com/twentyhq/twenty/issues/883) 

# Current Behavior: 

Currently, phone numbers appear with regular underlined formatting like other links on the platform and email addresses as an ordinary value. When clicking on a phone number to modify it, a standard input field opens for editing.

Image

# Desired Behavior: 

**1.	Design Enhancement:**
o	Phone and email links should be underlined, color = border/stronger , differentiating them from other links for better visual clarity
o	While hovering email links should also be underlined, color = text/primary 
o	While hovering phone links should also be underlined, color = text/primary and a pen should appear in the cell at the right

https://user-images.githubusercontent.com/102394218/257597980-9e449d52-151d-4d69-86ab-84a643fd55f9.mp4

**2.	Improved Interaction for Phone Edit:**
o	When click on the edit icon (pen) to modify a phone number, an overlay should open with additional options.
o	The overlay should allow users to change the country code associated with the phone number for improved flexibility.
o	Additionally, a button to copy the phone number should be available in the top right corner of the overlay for easier access.

https://user-images.githubusercontent.com/102394218/257598842-24fc58a3-a546-4a8c-86f7-76754c0b3c57.mp4

By implementing these changes, we aim to enhance the user experience, making it easier for users to interact with and modify phone and email links, and providing a clearer visual distinction between different types of links.

https://www.figma.com/file/xt8O9mFeLl46C5InWwoMrN/Twenty?type=design&node-id=568-41746&mode=design&t=rPOh4WY5St9iTCHU-4

```[tasklist]
### Tasks
- [ ] Introduce a new ContactLink that will behave as described above (underline + hover), use this link in EditablePhoneCell, create and EditableEmailCell and use it there too
- [ ] Create a PhoneInput in @/ui/input that will use https://github.com/jackocnr/intl-tel-input
- [ ] Style the component according to design
- [ ] Ignore the copy phone number button for now
```

### PR DESCRIPTION
**Pull request by mustajab-ikram**
closes #1055
image

cc: @charlesBochet

### CODE REVIEW

**Review by charlesBochet — COMMENTED** [review]
Looks good for now regarding the input, cells and fields :) 
The tricky part is about adding int-tel-input. Could you also add stories on top of Inputs so we can see them in storybook

**Comment by mustajab-ikram** [comment]
> Looks good for now regarding the input, cells and fields :) The tricky part is about adding int-tel-input. Could you also add stories on top of Inputs so we can see them in storybook

Thank you for your feedback, @charlesBochet! I'm glad to hear that the input, cells, and fields are looking good. I'll definitely take your note about 'int-tel-input' into consideration and work on adding the necessary implementation. I'll also get to work on creating the stories for the Inputs so that they can be showcased in Storybook.

**Comment by charlesBochet** [comment]
@mustajab-ikram Any update on this one? We try to not leave any PR open for a long time to keep the PRs list clean!
I'll close this one for now, but feel free to re-open it whenever you have progress :)

**Comment by charlesBochet** [comment]
I saw your message on Discord, re-opening as you're actively working on it!

File: front/src/modules/people/components/__stories__/PhoneEditableField.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { BrowserRouter } from 'react-router-dom';
+import type { Meta, StoryObj } from '@storybook/react';
+
+import { PhoneCellEdit } from '@/ui/table/editable-cell/type/components/PhoneCellEdit';
```
Comment (charlesBochet) [inline front/src/modules/people/components/__stories__/PhoneEditableField.stories.tsx:4]: @mustajab-ikram it's great that you are adding test. As this component is located in @/ui/table/editable-cell, storybook tests should be in UI/Table/Editable-Cell

File: front/src/modules/people/components/__stories__/PhoneEditableField.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import { ComponentDecorator } from '~/testing/decorators/ComponentDecorator';
+
+const meta: Meta<typeof PhoneCellEdit> = {
+  title: 'Modules/People/EditableFields/PhoneEditableField',
+  component: PhoneCellEdit,
+  decorators: [
+    (Story) => (
```
Comment (charlesBochet) [inline front/src/modules/people/components/__stories__/PhoneEditableField.stories.tsx:11]: why is Browser required to test this component, I feel ComponentDecorator is enough

File: front/src/modules/ui/input/email/components/__stories__/EmailInputDisplay.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import { ComponentDecorator } from '~/testing/decorators/ComponentDecorator';
+
+import { EmailInputDisplay } from '../EmailInputDisplay';
+
+const meta: Meta = {
+  title: 'Modules/People/EmailInputDisplay',
+  component: EmailInputDisplay,
+  decorators: [
+    (Story) => (
+      <StyledTestEmailContainer>
```
Comment (charlesBochet) [inline front/src/modules/ui/input/email/components/__stories__/EmailInputDisplay.stories.tsx:15]: same here, I don't think you need this decorator

File: front/src/modules/ui/input/phone/components/__stories__/PhoneInputDisplay.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+import { ComponentDecorator } from '~/testing/decorators/ComponentDecorator';
+
+import { PhoneInputDisplay } from '../PhoneInputDisplay'; // Adjust the import path as needed
+
+const meta: Meta = {
+  title: 'Modules/People/PhoneInputDisplay',
+  component: PhoneInputDisplay,
+  decorators: [
+    (Story) => (
+      <StyledTestPhoneContainer>
```
Comment (charlesBochet) [inline front/src/modules/ui/input/phone/components/__stories__/PhoneInputDisplay.stories.tsx:15]: same!

File: front/src/modules/ui/table/editable-cell/type/components/GenericEditableDoubleTextCellEditMode.tsx [hunk 5]
```diff
@@ -28,7 +28,7 @@ export function GenericEditableDoubleTextCellEditMode({ viewField }: OwnProps) {
   const [secondValue, setSecondValue] = useRecoilState<string>(
     tableEntityFieldFamilySelector({
       entityId: currentRowEntityId ?? '',
-      fieldName: viewField.metadata.firstValueFieldName,
+      fieldName: viewField.metadata.secondValueFieldName,
```
Comment (charlesBochet) [inline front/src/modules/ui/table/editable-cell/type/components/GenericEditableDoubleTextCellEditMode.tsx:31]: good catch

**Review by charlesBochet — APPROVED** [review]
This is good work. There are a few things that needed to be improve:
- storybook components decorator (see comment)
- storybook components location (see comment)
- also, the country indicator is not persisted, it should be, and then retrieved.

I'm going to merge it as it is and work on a follow up PR that I will link here

**Comment by charlesBochet** [comment]
Here it is: https://github.com/twentyhq/twenty/pull/1316/files

**Pull request merged**